File: tests/python/arrays/subarray_sort/test_subarray_sort.py
import pytest

from problems.arrays.subarray_sort.python.subarray_sort import subarray_sort
from tests.python.utility import load_test_cases


def test_subarray_sort(visualize: bool = False) -> None:

    test_cases = load_test_cases("arrays/subarray_sort.json")

    for idx, test in enumerate(test_cases):
        array = test["array"]
        expected = test["expected"]
        result = subarray_sort(array)

        assert (
            result == expected
        ), f"Test case {idx} failed. Expected: {expected}, got: {result}"

File: tests/cpp/arrays/test_subarray_sort.cpp
#include "../../../problems/arrays/subarray_sort/cpp/subarray_sort.h"
#include "../jsontestbase.h"

#include <vector>

class SubarraySortTest : public JsonTestBase {
public:
  SubarraySortTest() {
    json_file_path = std::string(TEST_CASES_DIR) + "/arrays/subarray_sort.json";
  }
};

TEST_F(SubarraySortTest, TestCases) {
  ASSERT_FALSE(test_cases.empty()) << "Test cases are empty";
  for (const auto &test : test_cases) {
    auto array = test["array"].get<std::vector<int>>();
    auto expected = test["expected"].get<std::vector<int>>();

    auto result = subarraySort(array);

    EXPECT_EQ(result, expected) << "Failed for test cases: " << test.dump(2);
  }
}
